### PR TITLE
Bugfix/lattice 1890 block writes to audit entity sets

### PR DESCRIPTION
**Pull request by katherinebernstein**

### CODE REVIEW

File: src/main/java/com/openlattice/datastore/data/controllers/DataIntegrationController.java [hunk 1]
```diff
@@ -45,8 +46,8 @@
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
-import static com.openlattice.authorization.EdmAuthorizationHelper.WRITE_PERMISSION;
-import static com.openlattice.authorization.EdmAuthorizationHelper.aclKeysForAccessCheck;
+import static com.openlattice.authorization.EdmAuthorizationHelper.*;
```
Comment (makosblade) [inline src/main/java/com/openlattice/datastore/data/controllers/DataIntegrationController.java:49]: 🤔

File: src/main/java/com/openlattice/datastore/data/controllers/DataIntegrationController.java [hunk 2]
```diff
@@ -215,17 +207,16 @@ public IntegrationResults integrateEntityAndAssociationData(
     @Override
     @PutMapping( "/" + EDGES )
     public int createEdges( @RequestBody Set<DataEdgeKey> associations ) {
-        final var entitySetIdChecks = new HashMap<AclKey, EnumSet<Permission>>();
+        final Set<UUID> entitySetIds = Sets.newHashSet();
```
Comment (makosblade) [inline src/main/java/com/openlattice/datastore/data/controllers/DataIntegrationController.java]: `withExpectedSize( associations.size * 3 );` perhaps?

File: src/main/java/com/openlattice/datastore/data/controllers/DataIntegrationController.java [hunk 3]
```diff
@@ -247,5 +238,36 @@ public int createEdges( @RequestBody Set<DataEdgeKey> associations ) {
         return propertyTypesByEntitySet;
     }
 
+    private Set<UUID> performAccessChecksOnEntitiesAndAssociations(
+            Set<Association> associations,
+            Set<Entity> entities ) {
+        final Set<UUID> entitySetIds = Sets.newHashSet();
```
Comment (makosblade) [inline src/main/java/com/openlattice/datastore/data/controllers/DataIntegrationController.java]: `withExpectedSize( (associations.size() * 3) + entitySetIds.size() );`

File: src/main/java/com/openlattice/datastore/data/controllers/DataController.java [hunk 4]
```diff
@@ -981,4 +991,41 @@ private static OffsetDateTime getDateTimeFromLong( long epochTime ) {
...
+        accessCheck( entitySetIds.stream().collect( Collectors.toMap( AclKey::new, id -> permissions ) ) );
+    }
+
+    private void ensureEntitySetCanBeWritten( UUID entitySetId ) {
+        ensureEntitySetsCanBeWritten( ImmutableSet.of( entitySetId ) );
+    }
+
+    private void ensureEntitySetsCanBeWritten( Set<UUID> entitySetIds ) {
+        if ( entitySetService.entitySetsContainFlag( entitySetIds, EntitySetFlag.AUDIT ) ) {
+            throw new ForbiddenException( "You cannot modify data of entity sets " + entitySetIds.toString()
```
Comment (makosblade) [inline src/main/java/com/openlattice/datastore/data/controllers/DataController.java]: I would think we'd want to differentiate entity sets that do have the flag vs those that don't but maybe that's TMI

**Review by makosblade — CHANGES_REQUESTED** [review]
Size sets when possible please 😄

File: src/main/java/com/openlattice/datastore/data/controllers/DataIntegrationController.java (hunk 1 above)
Comment (katherinebernstein) [reply src/main/java/com/openlattice/datastore/data/controllers/DataIntegrationController.java:49]: my autoformatter does this

File: src/main/java/com/openlattice/datastore/data/controllers/DataController.java (hunk 4 above)
Comment (geekbeast) [reply src/main/java/com/openlattice/datastore/data/controllers/DataController.java]: +1

File: src/main/java/com/openlattice/datastore/data/controllers/DataIntegrationController.java (hunk 1 above)
Comment (makosblade) [reply src/main/java/com/openlattice/datastore/data/controllers/DataIntegrationController.java:49]: Oh, I just made this face b/c `READ_PERMISSION` is being imported individually despite the `.*` import

Comment (katherinebernstein) [reply src/main/java/com/openlattice/datastore/data/controllers/DataIntegrationController.java:49]: oh wut. it's gone now.

**Review by makosblade — APPROVED** [review]

**Review by you-leee — APPROVED** [review]

**Review by pckeyes — APPROVED** [review]

**Pull request merged**